src/cr/cube/dimension.py

```diff
@@ -626,7 +626,24 @@ def _elements(self):
     @lazyproperty
     def _elements_transforms(self):
         """Element transform dict expressed in the dimension transforms expression."""
-        return self._dimension_transforms_dict.get("elements", {})
+        transforms = self._dimension_transforms_dict.get("elements", {})
+
+        # ---include MR insertions into "hide" consideration
+        if self._dimension_type == DT.MR:
```

Comment:
Should this be `in DT.ARRAY_TYPES` for future-proofing?  I know we only support MR for now, but I thought that numeric arrays would also require these derived zz9 insertions, and may not be too far behind. (This would allow cat array subvars too, but on the subvars dim, we would need derived insertions too, right?)

Reply:
refactored